### PR TITLE
allow for GR client to be configured on instantiation

### PR DESCRIPTION
**Pull request by soberstadt**

### CODE REVIEW

**Comment by Omicron7** [comment]
:+1: @twinge You want to check this at all?

File: lib/global_registry/base.rb [hunk 1]
```diff
@@ -4,26 +4,46 @@
 
 module GlobalRegistry
   class Base
+    def initialize(args = {})
+      @base_url = args[:base_url]
+      @access_token = args[:access_token]
+      @xff = args[:xff]
+    end
 
     def self.find(id, params = {})
```
Comment (twinge) [inline lib/global_registry/base.rb:13]: The class level methods should probably all be rewritten to use the instance level methods now. E.g:

``` ruby
def self.find(id, params = {})
  new.find(id, params)
end
```

**Comment by soberstadt** [comment]
@twinge how's that?

**Comment by twinge** [comment]
:+1:

**Pull request merged**